lib/auth/auth_with_roles.go

```diff
@@ -623,16 +623,22 @@ func (a *ServerWithRoles) GetCertAuthority(ctx context.Context, id types.CertAut
 		readVerb = types.VerbRead
 	}
 
-	ca, err := a.authServer.GetCertAuthority(ctx, id, loadKeys, opts...)
+	// Create a minimum CA for calculating access to the cert authority.
+	contextCA, err := types.NewCertAuthority(types.CertAuthoritySpecV2{
+		Type:        id.Type,
+		ClusterName: id.DomainName,
+	})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	sctx := &services.Context{User: a.context.User, Resource: ca}
+
+	sctx := &services.Context{User: a.context.User, Resource: contextCA}
 	if err := a.actionWithContext(sctx, apidefaults.Namespace, types.KindCertAuthority, readVerb); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	return ca, nil
+	ca, err := a.authServer.GetCertAuthority(ctx, id, loadKeys, opts...)
```

Comment:
Why don't we also need to call `actionWithContext` on this specific `ca`?

Reply:
At the moment, the only things the auth checker checks for are the name and CA type, so it wouldn't add much here. If we add it in the future, however, it could be of use. I'm hesitant to add it unless we need to though. Thoughts?

Reply:
Thinking on this, it seems like a good idea to add. I've added this in.